In [12]:
import sys
from pathlib import Path

sys.path.append(str(Path().resolve().parent.parent))

In [13]:


import pandas as pd
from src.utils.data_check import DataCheck

In [15]:
df = pd.read_csv("../../data/raw/austo_automobile1.csv")

In [16]:
overview = DataCheck.overview(df)

In [17]:
print("Shape:", overview["shape"])

Shape: (1581, 14)


In [19]:
print("\nMissing values:\n", overview["missing_values"])


Missing values:
 Age                 0
Gender              0
Profession          0
Marital_status      0
Education           0
No_of_Dependents    0
Personal_loan       0
House_loan          0
Partner_working     0
Salary              0
Partner_salary      0
Total_salary        0
Price               0
Make                0
dtype: int64


In [20]:
print("\nMissing percentage:\n", overview["missing_percentage"])


Missing percentage:
 Age                 0.0
Gender              0.0
Profession          0.0
Marital_status      0.0
Education           0.0
No_of_Dependents    0.0
Personal_loan       0.0
House_loan          0.0
Partner_working     0.0
Salary              0.0
Partner_salary      0.0
Total_salary        0.0
Price               0.0
Make                0.0
dtype: float64


In [21]:
print("\nDuplicate rows:", overview["duplicate_rows"])


Duplicate rows: 0


In [22]:
print("\nData types:\n", overview["dtypes"])


Data types:
 Age                  int64
Gender              object
Profession          object
Marital_status      object
Education           object
No_of_Dependents     int64
Personal_loan       object
House_loan          object
Partner_working     object
Salary               int64
Partner_salary       int64
Total_salary         int64
Price                int64
Make                object
dtype: object


In [24]:
display(DataCheck.summary_stat(df))

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Age,1581.0,NaN,NaN,NaN,32.211259,9.125477,22.0,25.0,29.0,38.0,60.0
Gender,1581,2,Male,1252,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Profession,1581,2,Salaried,896,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Marital_status,1581,2,Married,1443,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Education,1581,2,Post Graduate,985,NaN,NaN,NaN,NaN,NaN,NaN,NaN
No_of_Dependents,1581.0,NaN,NaN,NaN,2.457938,0.943483,0.0,2.0,2.0,3.0,4.0
Personal_loan,1581,2,Yes,792,NaN,NaN,NaN,NaN,NaN,NaN,NaN
House_loan,1581,2,No,1054,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Partner_working,1581,2,Yes,868,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Salary,1581.0,NaN,NaN,NaN,59732.447818,14278.642665,30000.0,51000.0,59000.0,71000.0,90000.0


In [25]:
display(DataCheck.unique_values(df))

Age                  39
Gender                2
Profession            2
Marital_status        2
Education             2
No_of_Dependents      5
Personal_loan         2
House_loan            2
Partner_working       2
Salary               61
Partner_salary       19
Total_salary        123
Price                53
Make                  3
dtype: int64

- The dataset has 1581 rows and 14 columns.
- There are no missing values.
- There are no duplicate rows.
- Total_salary is a derived field and should equal Salary + Partner_salary.